### PR TITLE
feat(kuma-dp): publish metrics with text_readouts from envoy

### PR DESCRIPTION
**Pull request by Automaat**
When publishing metrics with text_readouts we can add grafana dashboard to dataplane view with information to which CP this DP was connected at given time, it is especially useful when debugging.

Fixes: #1011
Signed-off-by: Marcin Skalski <marcin.skalski@konghq.com>

### Checklist prior to review

- [ ] Link to docs PR or issue --
- [x] Link to UI issue or PR --
- [x] Is the [issue worked on linked][1]? --
- [x] The PR does not hardcode values that might break projects that depend on kuma (e.g. "kumahq" as a image registry) --
- [x] The PR will work for both Linux and Windows, system specific functions like `syscall.Mkfifo` have equivalent implementation on the other OS --
- [x] Unit Tests --
- [ ] E2E Tests --
- [x] Manual Universal Tests --
- [x] Manual Kubernetes Tests --
- [x] Do you need to update [`UPGRADE.md`](../blob/master/UPGRADE.md)? --
- [x] Does it need to be backported according to the [backporting policy](../blob/master/CONTRIBUTING.md#backporting)? --
- [x] Do you need to explicitly set a [`> Changelog:` entry here](../blob/master/CONTRIBUTING.md#submitting-a-patch) or add a `ci/` label to run fewer/more tests?

[1]: https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword

### CODE REVIEW

**Comment by Automaat** [comment]
This is how it looks in grafana, just this metric: 
Screenshot 2022-10-17 at 15 45 50
How it looks in dp dashboard:
Screenshot 2022-10-17 at 15 45 40

File: app/kumactl/data/install/k8s/metrics/grafana/kuma-dataplane.json [hunk 1]
```diff
@@ -1961,6 +2060,7 @@
       "steppedLine": false,
       "targets": [
         {
+          "datasource": "Prometheus",
```
Comment (lahabana) [inline app/kumactl/data/install/k8s/metrics/grafana/kuma-dataplane.json:2063]: this capital case change is strange no?

Comment (lahabana) [reply app/kumactl/data/install/k8s/metrics/grafana/kuma-dataplane.json:2063]: This entry is also line 2019 which is strange

File: app/kumactl/data/install/k8s/metrics/grafana/kuma-dataplane.json [hunk 2]
```diff
@@ -432,6 +432,105 @@
...
+          "expr": "envoy_control_plane_identifier{dataplane=\"$dataplane\", kuma_io_zone=~\"$zone\", mesh=\"$mesh\"}",
+          "format": "time_series",
+          "instant": false,
+          "interval": "",
+          "legendFormat": "{{text_value}}",
+          "range": true,
+          "refId": "A"
+        }
+      ],
+      "title": "Connection to specific Control Plane",
```
Comment (lahabana) [inline app/kumactl/data/install/k8s/metrics/grafana/kuma-dataplane.json:531]: I can't find the right naming for this.

I find it tricky to figure it's a timeseries. Is there anyway to put times on the x axis to make this clearer?

Also maybe rewording the title might help maybe something like:
```suggestion
      "title": "Control Plane connected to",
```

**Comment by lahabana** [comment]
@Automaat seems like some tests fail: app/kuma-dp/pkg/dataplane/metrics

File: app/kumactl/data/install/k8s/metrics/grafana/kuma-dataplane.json (hunk 2 above)
Comment (Automaat) [reply app/kumactl/data/install/k8s/metrics/grafana/kuma-dataplane.json:531]: I was not happy with this name from the begging, but couldn't find anything better. I think your suggestion is better. As for the time on x axis, I thought I was missing something, and I now know what. Thanks, I will add your suggestions

File: app/kumactl/data/install/k8s/metrics/grafana/kuma-dataplane.json (hunk 1 above)
Comment (Automaat) [reply app/kumactl/data/install/k8s/metrics/grafana/kuma-dataplane.json:2063]: It is strange, but it is what Grafana produced, so I decided I won't argue with it. But I can revert it if you want

Comment (lahabana) [reply app/kumactl/data/install/k8s/metrics/grafana/kuma-dataplane.json:2077]: No it's fine if it's coming from Grafana :)

**Comment by Automaat** [comment]
This is how this panel looks like after changes: 
Screenshot 2022-10-18 at 15 54 42

**Review by lahabana — APPROVED** [review]

**Review by jakubdyszkiewicz — APPROVED** [review]
